inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/metrics/MeasuresAndViewsManager.java

```diff
@@ -0,0 +1,298 @@
+                    addOrUpdateAndCacheMeasureWithViews(defWithDefaults, registeredMeasures, registeredViews);
+                }
+            });
+        }
+        //TODO: delete removed measures and views as soon as this is possible in Open-Census
+    }
+
+    /**
+     * Tries to create a measure based on the given definition as well as its views.
+     * If the measure or a view already exists, info messages are printed out
+     *
+     * @param definition         the definition of the measure and its views. The defaults
+     *                           must be already populated using {@link MetricDefinitionSettings#getCopyWithDefaultsPopulated(String)}!
+     * @param registeredMeasures a map of which measures are already registered at the OpenCensus API. Maps the names to the measures. Used to detect if the measure to create already exists.
+     * @param registeredViews    same as registeredMeasures, but maps the view names to the corresponding registered views.
+     */
+    @VisibleForTesting
+    void addOrUpdateAndCacheMeasureWithViews(MetricDefinitionSettings definition, Map<String, Measure> registeredMeasures, Map<String, View> registeredViews) {
+        try {
+            String measureName = definition.getName();
+            Measure measure = registeredMeasures.get(measureName);
+            if (measure != null) {
+                updateMeasure(measure, definition);
+            } else {
+                measure = createNewMeasure(definition);
+            }
+            val resultMeasure = measure;
+
+            val metricViews = definition.getViews();
+            metricViews.forEach((name, view) -> {
+                if (view.isEnabled()) {
+                    try {
+                        addAndRegisterOrUpdateView(resultMeasure, view, registeredViews);
+                    } catch (Exception e) {
+                        log.error("Error creating view '{}'!", name, e);
+                    }
+                }
+            });
+
+            //TODO: delete views which where created by this class but have been removed from the given metric as soon as OpenCensus supports it
+            definitionsUsedToDefineMetrics.put(measureName, definition);
+            cachedMeasures.put(measureName, measure);
+
+        } catch (Exception e) {
+            log.error("Error creating metric", e);
+        }
+    }
+
+    private Measure createNewMeasure(MetricDefinitionSettings fullDefinition) {
+        Measure measure;
+        switch (fullDefinition.getType()) {
+            case LONG:
+                measure = Measure.MeasureLong.create(fullDefinition.getName(), fullDefinition.getDescription(), fullDefinition.getUnit());
+                break;
+            case DOUBLE:
+                measure = Measure.MeasureDouble.create(fullDefinition.getName(), fullDefinition.getDescription(), fullDefinition.getUnit());
+                break;
+            default:
+                throw new RuntimeException("Unhandled case!");
```

Comment:
Let's use a more specific message "The type '{}' is not supported for measurements."